compositor_pipeline/src/pipeline/encoder/ffmpeg_h264.rs

```diff
@@ -0,0 +1,295 @@
+                //  Maximum motion vector search range - libx264 defaults to 16 (in case of placebo
+                //  or veryslow preset to 24)
+                ("me_range", "16"),
+                // Max QP step - libx264 defaults to 4
+                ("qdiff", "4"),
+                // Min QP - libx264 defaults to 0
+                ("qmin", "0"),
+                // Max QP - libx264 defaults to QP_MAX = 69
+                ("qmax", "69"),
+                //  Maximum GOP (Group of Pictures) size - libx264 defaults to 250
+                ("g", "250"),
+                // QP factor between I and P frames - libx264 defaults to 1.4 (in case of tune=grain to 1.1)
+                ("i_qfactor", "1.4"),
+                // QP factor between P and B frames - libx264 defaults to 1.4 (in case of tune=grain to 1.1)
+                ("f_pb_factor", "1.3"),
+                // A comma-separated list of partitions to consider. Possible values: p8x8, p4x4, b8x8, i8x8, i4x4, none, all
+                ("partitions", options.preset.default_partitions()),
+                // Subpixel motion estimation and mode decision (decision quality: 1=fast, 11=best)
+                ("subq", options.preset.default_subq_mode()),
+            ]),
+        )?;
+
+        result_sender.send(Ok(())).unwrap();
+
+        loop {
+            let frame = match frame_receiver.recv() {
+                Ok(Message::Frame(f)) => f,
+                Ok(Message::Stop) => break,
+                Err(_) => break,
+            };
+
+            if frame_receiver.len() > 20 {
+                warn!("Dropping frame: render queue is too long.",);
+                continue;
+            }
+
+            let mut av_frame = frame::Video::new(
+                Pixel::YUV420P,
+                options.resolution.width as u32,
+                options.resolution.height as u32,
+            );
+
+            frame_into_av(frame, &mut av_frame);
+
+            if let Err(e) = encoder.send_frame(&av_frame) {
+                error!("Encoder error: {e}.")
+            }
+
+            let mut packet = Packet::empty();
+
+            while encoder.receive_packet(&mut packet).is_ok() {
+                match EncodedChunk::from_av_packet(&packet, EncodedChunkKind::Video(Codec::H264)) {
+                    Ok(chunk) => {
+                        packet_sender.send(chunk).unwrap();
+                    }
+                    Err(e) => {
+                        warn!("failed to parse an ffmpeg packet received from encoder: {e}",);
+                        break;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Drop for LibavH264Encoder {
+    fn drop(&mut self) {
+        self.frame_sender.send(Message::Stop).unwrap();
+        match self.encoder_thread.take() {
+            Some(handle) => handle.join().unwrap(),
+            None => error!(
+                "[output {}] Encoder thread was already joined. Something very bad has happened.",
```

Comment:
>  Something very bad has happened.

That does not look good in an error message. In cases where some condition is clearlly a bug use phrases like
- This should not/never happen.
- Internal compositor error
- This is a bug, please report it. (I'm not fan of this one, but saw it in some places)